File: ui/src/models/ClubMessage.ts
export type ClubMessageCandidate = {
    user_id: string;
    display_name: string;
    email: string;
};

export type ClubChatSummary = {
    chat_id: string;
    club_id: string;
    other_user_id: string;
    other_display_name: string;
    other_email: string;
    last_message: string;
    last_sender_user_id: string;
    last_message_at: string;
    unread_count: number;
};

export type UnreadByClub = {
    clubId: string;
    clubName: string;
    unread: number;
};

export type UnreadSummary = {
    total: number;
    byClub: UnreadByClub[];
};

export type ClubChatMessage = {
    id: string;
    chat_id: string;
    sender_user_id: string;
    sender_display_name: string;
    sender_email: string;
    content: string;
    created_at: string;
};

export type ClubChatCreated = {
    chat_id: string;
};

File: api_go/models/ClubMessage.go
package models

import "time"

type ClubMessageCandidate struct {
	UserID      string
	DisplayName string
	Email       string
}

type ClubChatSummary struct {
	ChatID           string
	ClubID           string
	OtherUserID      string
	OtherDisplayName string
	OtherEmail       string
	LastMessage      string
	LastSenderUserID string
	LastMessageAt    *time.Time
	UnreadCount      int
}

type UnreadByClub struct {
	ClubID   string
	ClubName string
	Unread   int
}

type UnreadSummary struct {
	Total  int
	ByClub []UnreadByClub
}

type ClubChatMessage struct {
	ID                string
	ChatID            string
	SenderUserID      string
	SenderDisplayName string
	SenderEmail       string
	Content           string
	CreatedAt         *time.Time
}
